### PR TITLE
Refactor odh-e2e to use test-frame and avoid duplication of code

### PR DESCRIPTION
**Pull request by kornys**

### CODE REVIEW

File: src/test/java/io/odh/test/e2e/upgrade/OlmUpgradeST.java [hunk 1]
```diff
@@ -99,14 +99,15 @@ void testUpgradeOlm() throws IOException, InterruptedException {
 
         LOGGER.info("Upgrade to next available version in OLM catalog");
         // Approve upgrade to newer version
-        KubeUtils.waitForInstallPlan(olmInstall.getNamespace(), olmInstall.getCsvName());
-        ip = ResourceManager.getKubeClient().getNonApprovedInstallPlan(olmInstall.getNamespace(), olmInstall.getCsvName());
-        ResourceManager.getKubeClient().approveInstallPlan(olmInstall.getNamespace(), ip.getMetadata().getName());
+        KubernetesUtils.waitForInstallPlan(olmInstall.getNamespace(), olmInstall.getCsvName());
+        ip = io.skodjob.testframe.utils.KubeUtils.getNonApprovedInstallPlan(olmInstall.getNamespace(), olmInstall.getCsvName());
+        io.skodjob.testframe.utils.KubeUtils.approveInstallPlan(olmInstall.getNamespace(), ip.getMetadata().getName());
```
Comment (jiridanek) [inline src/test/java/io/odh/test/e2e/upgrade/OlmUpgradeST.java:104]: Import `io.skodjob.testframe.utils.KubeUtils` in this file?

File: src/test/java/io/odh/test/e2e/upgrade/OlmUpgradeST.java [hunk 2]
```diff
@@ -99,14 +99,15 @@ void testUpgradeOlm() throws IOException, InterruptedException {
 
         LOGGER.info("Upgrade to next available version in OLM catalog");
         // Approve upgrade to newer version
-        KubeUtils.waitForInstallPlan(olmInstall.getNamespace(), olmInstall.getCsvName());
-        ip = ResourceManager.getKubeClient().getNonApprovedInstallPlan(olmInstall.getNamespace(), olmInstall.getCsvName());
-        ResourceManager.getKubeClient().approveInstallPlan(olmInstall.getNamespace(), ip.getMetadata().getName());
+        KubernetesUtils.waitForInstallPlan(olmInstall.getNamespace(), olmInstall.getCsvName());
```
Comment (jiridanek) [inline src/test/java/io/odh/test/e2e/upgrade/OlmUpgradeST.java:102]: Having both `KubernetesUtils` and `KubeUtils` seems somewhat confusing.

File: src/test/java/io/odh/test/e2e/upgrade/OlmUpgradeST.java (hunk 1 above)
Comment (kornys) [reply src/test/java/io/odh/test/e2e/upgrade/OlmUpgradeST.java:104]: It is draft...

File: src/test/java/io/odh/test/e2e/upgrade/OlmUpgradeST.java (hunk 2 above)
Comment (jiridanek) [reply src/test/java/io/odh/test/e2e/upgrade/OlmUpgradeST.java:102]: I did get approve for a draft here in the past, even :D Just now I felt like writing down one or two thoughts. Feel free to ignore, ofc 😉

**Comment by kornys** [comment]
@Frawless @jiridanek 
It is ready for review (I do not plan to merge it now). 
Test-frame is used from GH snapshot releases.
Otherwise it should work, Im running tests in jenkins now

File: src/main/java/io/odh/test/framework/manager/resources/DataScienceClusterResource.java [hunk 3]
```diff
@@ -114,7 +132,8 @@ public boolean waitForReadiness(DataScienceCluster resource) {
             // https://github.com/red-hat-data-services/rhods-operator/blob/rhoai-2.8/controllers/datasciencecluster/datasciencecluster_controller.go#L257
 
             // Wait for standard Kubernetes condition types (status for the whole DSC)
-            record ConditionExpectation(String conditionType, String expectedStatus) { }
+            record ConditionExpectation(String conditionType, String expectedStatus) {
+            }
```
Comment (Frawless) [inline src/main/java/io/odh/test/framework/manager/resources/DataScienceClusterResource.java:136]:
```suggestion
            record ConditionExpectation(String conditionType, String expectedStatus) { }
```

File: src/main/java/io/odh/test/framework/manager/resources/DataScienceClusterResource.java [hunk 4]
```diff
@@ -146,20 +165,25 @@ record ConditionExpectation(String conditionType, String expectedStatus) { }
             dscReady = dscReady && hasCreationSuccessfulEvent;
 
             return dscReady;
-        }, () -> { });
+        }, () -> {
+        });
```
Comment (Frawless) [inline src/main/java/io/odh/test/framework/manager/resources/DataScienceClusterResource.java:169]:
```suggestion
        }, () -> { });
```

**Comment by kornys** [comment]
Both jenkins jobs for odh-e2e passed with these changes

File: pom.xml [hunk 5]
```diff
@@ -63,6 +63,7 @@
         <aspectj.version>1.9.21.2</aspectj.version>
         <allure.version>2.25.0</allure.version>
         <allure.maven.version>2.12.0</allure.maven.version>
+        <test-frame.version>0.1.0-SNAPSHOT</test-frame.version>
```
Comment (jiridanek) [inline pom.xml:66]: snapshot is here to stay?

File: pom.xml [hunk 6]
```diff
@@ -240,7 +262,7 @@
                 </executions>
                 <configuration>
                     <skipITs>${it.skip}</skipITs>
-                    <forkCount>0</forkCount>
+                    <forkCount>1</forkCount>
```
Comment (jiridanek) [inline pom.xml:265]: this is necessary, or just as a good idea?

**Review by jiridanek — APPROVED** [review]
One thing that does not work for me. In IntelliJ, when I click Download Sources, I get

> Sources not found for: io.skodjob:test-frame-common:0.1.0-20240507.094145-86

And general observation that

```
KubeResourceManager.getInstance().createResourceWithWait(dsci)
```

is informative, but quite long ;P

File: pom.xml (hunk 6 above)
Comment (kornys) [reply pom.xml:265]: it is recommended to use forkCount > 0, also it allows us to spec env vars in failsafe plugin

File: pom.xml (hunk 5 above)
Comment (kornys) [reply pom.xml:66]: no :) I will switch that once I release version with stable changes, beta3 I would say

Comment (jiridanek) [reply pom.xml:66]:
```
 Error:  COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
Error:  /home/runner/work/odh-e2e/odh-e2e/src/test/java/io/odh/test/e2e/Abstract.java:[49,42] cannot find symbol
  symbol:   method addCreateCallback((r)->{ if [...]; } })
  location: class io.skodjob.testframe.resources.KubeResourceManager
```

in gha tests, that's something wrong in the test-frame lib itself? I can find no references to `addCreateCallback` here.

**Comment by kornys** [comment]
> One thing that does not work for me. In IntelliJ, when I click Download Sources, I get
> 
> > Sources not found for: io.skodjob:test-frame-common:0.1.0-20240507.094145-86
> 
> And general observation that
> 
> ```
> KubeResourceManager.getInstance().createResourceWithWait(dsci)
> ```
> 
> is informative, but quite long ;P

Yeah it is because to github snapshots we do not release sources. So once we switch to maven central when I release next beta release, sources will be available

File: pom.xml (hunk 5 above)
Comment (kornys) [reply pom.xml:66]: fixed, due to snaphot was cachet I had to remove cache in GH action to download latest snapshot :)

Comment (jiridanek) [reply pom.xml:66]: either that, or maybe give it `--update-snapshots` for now?

Comment (kornys) [reply pom.xml:66]: I don not expect merge today :) I will release tomorrow or later this week, then switch to maven central here and then merge :)

**Pull request merged**